citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/DbTokenStore.java

```diff
@@ -184,4 +184,13 @@ private String extractTokenKey(String value) {
             }
         }
     }
+    
+    public List<DbAccessToken> findTokensByUserName(String userName) {
+        return (List<DbAccessToken>)dbAccessTokenRepository.findByUsername(userName);
+    }
+    
+    public void revokeAccessToken(String clientId) {
+        if(clientId != null) 
+            dbAccessTokenRepository.deleteByClientId(clientId);
```

Comment:
that would revoke access for *all* users for that app, wouldn't it? That does not make sense. I don't want you to revoke access of an app to my account.